linalg/sparsemat.hpp

```diff
@@ -81,23 +95,23 @@ class SparseMatrix : public AbstractSparseMatrix
    void Destroy();   // Delete all owned data
    void SetEmpty();  // Init all entries with empty values
 
-   bool useCuSparse{true}; // Use cuSPARSE if available
+   bool useGPUSparse{true}; // Use cuSPARSE or hipSPARSE if available
 
-   // Initialize cuSPARSE
-   void InitCuSparse();
+   // Initialize cuSPARSE/hipSPARSE
+   void InitGPUSparse();
 
-#ifdef MFEM_USE_CUDA
-   cusparseStatus_t status;
-   static cusparseHandle_t handle;
-   cusparseMatDescr_t descr=0;
+#ifdef CUDA_or_HIP
+   cu_or_hip(sparseStatus_t) status;
+   static cu_or_hip(sparseHandle_t) handle;
+   cu_or_hip(sparseMatDescr_t) descr=0;
```

Comment:
No need to use the macros here (so we don't need to make them public), just use
```c++
#ifdef MFEM_USE_CUDA
   // ...
#elif defined(MFEM_USE_HIP)
   // ...
#endif
```

Reply:
done